editor/find_in_files_scanner.cpp

```diff
@@ -0,0 +1,352 @@
+			++current_line;
+			offset = 0;
+		}
+
+		int repl_begin = locations[i].begin + offset;
+		int repl_end = locations[i].end + offset;
+
+		int _;
+		if (!find_next(line, search_text, repl_begin, match_case, whole_words, _, _)) {
+			// Make sure the replacement is still valid in case the file was tampered with.
+			print_verbose(String("Occurrence no longer matches, replace will be ignored in {0}: line {1}, col {2}").format(varray(file->get_path(), repl_line_number, repl_begin)));
+			continue;
+		}
+
+		line = line.left(repl_begin) + new_text + line.substr(repl_end);
+		// Keep an offset in case there are successive replaces in the same line.
+		offset += new_text.length() - (repl_end - repl_begin);
+	}
+
+	buffer += line;
+
+	while (!file->eof_reached()) {
+		buffer += conservative.get_line(file);
+	}
+
+	// Now the modified contents are in the buffer, rewrite the file with our changes.
+
+	Error err = file->reopen(file_path, FileAccess::WRITE);
+	ERR_FAIL_COND_MSG(err != OK, "Cannot create file in path '" + file->get_path() + "'.");
+
+	file->store_string(buffer);
+}
+
+bool DefaultFindInFilesScanner::find_next(const String &line, String pattern, int from, bool match_case, bool whole_words, int &out_begin, int &out_end) {
+	int end = from;
+
+	while (true) {
+		int begin = match_case ? line.find(pattern, end) : line.findn(pattern, end);
+
+		if (begin == -1) {
+			return false;
+		}
+
+		end = begin + pattern.length();
+		out_begin = begin;
+		out_end = end;
+
+		if (whole_words) {
+			if (begin > 0 && (is_ascii_identifier_char(line[begin - 1]))) {
+				continue;
+			}
+			if (end < line.size() && (is_ascii_identifier_char(line[end]))) {
+				continue;
+			}
+		}
+
+		return true;
+	}
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+void BuiltinScriptFindInFilesScanner::scan(Ref<FileAccess> file, String file_path, String search_text, bool match_case, bool whole_words, Vector<FindInFilesScanner::ScanMatch> &matches) {
+	int matches_offset = matches.size();
+	for (auto &match : _parse_tscn(file)) {
+		file->seek(0);
+		String relative_file_path = file_path + "::" + match.key;
+		DefaultFindInFilesScanner::scan(file, relative_file_path, match.value.display_text, search_text, match_case, whole_words, matches, Size2i(match.value.script_start_line, match.value.line_idx - 1));
+
+		// Let's handle some edge cases!
+		if (matches.size() > matches_offset) {
+			// We have new matches
+
+			// Check if the first new match is on the first line, if so handle it separately
+			if (matches.get(matches_offset).line_number == 1) {
+				FindInFilesScanner::ScanMatch scan_match = matches.get(matches_offset);
+				if (scan_match.begin - SCRIPT_SOURCE_TAG_LEN == scan_match.end - SCRIPT_SOURCE_TAG_LEN - 1) {
+					// Edge case matching the double quote mark
+					matches.remove_at(matches_offset);
+				} else {
+					// Another edge case where we need to remove "script/source = " from the resulting line
```

Comment:
Just to note it here, randomly, but our comment style is `// Message.`, and most of code comments in this PR are missing that period at the end.